src/components/graph-view.js

```diff
@@ -673,47 +673,59 @@ class GraphView extends React.Component<IGraphViewProps, IGraphViewState> {
     return !!GraphUtils.findParent(element, '.edge-container');
   }
 
-  handleNodeMove = (position: IPoint, nodeId: string, shiftKey: boolean) => {
-    const { canCreateEdge, readOnly, nodeKey } = this.props;
+  positionNodes = (
+    newPosition: IPoint,
+    movedNode: INode,
+    nodeMoving: boolean
+  ) => {
+    const { nodeKey } = this.props;
     const { selectedNodes } = this.state;
-    const nodeMapNode: INodeMapNode | null = this.getNodeById(nodeId);
+    const deltaX = newPosition.x - movedNode.x;
+    const deltaY = newPosition.y - movedNode.y;
 
-    if (!nodeMapNode) {
-      return;
-    }
+    const isSelected = selectedNodes.find(
+      node => node[nodeKey] === movedNode[nodeKey]
+    );
 
-    const node = nodeMapNode.node;
+    const nodesToPosition = isSelected
+      ? selectedNodes
+      : selectedNodes.concat([movedNode]);
```

Comment:
Just curious about the reason behind adding the movedNode to the selectedNodes? What's the use case for when this would happen?

Reply:
I just moved this logic around, but I believe the logic is if we attempt to move a node which hasn't yet been selected, that node *and* the selected nodes are moved.